twython/endpoints.py

```diff
@@ -199,7 +199,28 @@ def upload_video(self, media, media_type, size=None):
             'command': 'FINALIZE',
             'media_id': media_id
         }
-        return self.post(upload_url, params=params)
+        response_finalize = self.post(upload_url, params=params)
+
+        # Stage 4: STATUS call if still processing
+        params = {
+            'command': 'STATUS',
+            'media_id': media_id
+        }
+        
+        processing_state = response_finalize['processing_info'].get('state', None)
```

Comment:
You don't need to default to `None` as `processing_state = response_finalize['processing_info'].get('state')` will return `None` if it doesn't exist